foreshadow/concrete/internals/fancyimpute.py

```diff
@@ -95,5 +96,24 @@ def transform(self, X):
         Returns:
             :obj:`pandas.DataFrame`: Output data
 
+        Raises:
+            e: ValueError
+
         """
-        return self.imputer.complete(X)
+        try:
+            return self.imputer.complete(X)
+        except ValueError as e:
+            """This is a temporary fix since the newer version of
+            fancyimpute package has already fixed the issue of throwing
+            exception when there's no missing value. However, due to the
+            constraint on the requirements, we hacve to stage with an older
+            version and use this workaround until we figure out how to
+            upgrade all the associated dependencies.
+            """
+            if "Input matrix is not missing any values" in str(e):
```

Comment:
Instead of checking for the  error msg in the exception, what about checking that the dataframe has no missing values, so something like:
```
if X.notnull().all(axis=None):
  logging.info(...)
  return X
else:
  return self.imputer.complete(X)
```

Reply:
Done.